comm/src/main/scala/coop/rchain/p2p/p2p.scala

```diff
@@ -61,6 +61,18 @@ case class Network(homeAddress: String) extends ProtocolDispatcher[java.net.Sock
 
   val net = new UnicastNetwork(local, Some(this))
 
+  private val unresponsive = new scala.collection.mutable.HashMap[String,Long]
+
+  private def checkResponsive(key: String): Option[String] =
+    unresponsive.get(key) match {
+      case Some(timestamp) if timestamp < System.currentTimeMillis => {
```

Comment:
Is following object capability discipline a norm in RChain code?

Using `System.currentTimeMillis` here complicates unit testing etc. as usual when object capability discipline isn't followed.

Reply:
I don't know what that means. but it sounds like you disapprove of using things like the system clock for some reason. This is an older version of the PR you're reviewing. The newer version also uses the clock, though it's better hidden.

Is there an acceptable way to get at "wait five minutes"? I'm not schooled in object capabilities.

Reply:
Typically, the object would be passed a clock at construction time and you would call `myClock.nowInMillis()` or `myClock.fiveMinutesHence()` or whatever you want. Then your code doesn't require any ambient authority from the system, it just has whatever semantic authority comes along with the clock. In typical uses, the clock might still be the usual system clock, but it doesn't have to be. So, you can write tests where five minutes hence is right now. And, also you can provide it a clock that hides information about your location (e.g., by being in Zulu time and not localized).

Reply:
If you're following ocap discipline, any powerful references such as the clock get passed into the object as an argument on the constructor or some other method. This lets you substitute a mock clock for testing, for example.

Reply:
That all sounds dandy. I don't think I want this PR merged into dev at this point, since this isn't the way we're going to handle non-communicative nodes anyway. I'll ponder this capabilities business.